pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentStickyKeyDispatcherMultipleConsumers.java

```diff
@@ -219,6 +242,24 @@ protected synchronized boolean trySendMessagesToConsumers(ReadType readType, Lis
                     }
                 }
             }
+
+            // Update if the markDeletePosition move forward
+            updateIfNeededAndGetLastSentPosition();
+
+            // Should not access to individualDeletedMessages from outside managed cursor
+            // because it doesn't guarantee thread safety.
+            if (lastSentPosition == null) {
```

Comment:
Just mark this for easier reading: initialize `lastSentPosition & individuallySentPositions ` after the dispatcher created or the first consumer joined.